mods/auto-theme-switcher.wh.cpp

```diff
@@ -0,0 +1,377 @@
+            ApplyLockScreen();
+    }).detach();
+}
+
+SYSTEMTIME ParseScheduleTime(PCWSTR timeStr) {
+    SYSTEMTIME st = {};
+    swscanf_s(timeStr, L"%hu:%hu", &st.wHour, &st.wMinute);
+    return st;
+}
+
+time_t GetNextSwitch(const SYSTEMTIME& light, const SYSTEMTIME& dark, bool& nextLight) {
+    time_t now = time(nullptr);
+    struct tm local;
+    localtime_s(&local, &now);
+
+    auto makeTime = [&](const SYSTEMTIME& st) {
+        struct tm t = local;
+        t.tm_hour = st.wHour; t.tm_min = st.wMinute; t.tm_sec = 0;
+        return mktime(&t);
+    };
+
+    time_t lightT = makeTime(light);
+    time_t darkT = makeTime(dark);
+
+    bool isLightNow;
+    if (lightT < darkT)
+        isLightNow = now >= lightT && now < darkT;
+    else
+        isLightNow = now >= lightT || now < darkT;
+
+    if (isLightNow) {
+        nextLight = false;
+        if (darkT <= now) darkT += 86400;
+        return darkT;
+    } else {
+        nextLight = true;
+        if (lightT <= now) lightT += 86400;
+        return lightT;
+    }
+}
+
+DWORD WINAPI ThemeScheduler(LPVOID) {
+    while (!g_exitFlag) {
+        bool nextLight;
+        time_t now = time(nullptr);
+        time_t nextSwitch = GetNextSwitch(g_lightTime, g_darkTime, nextLight);
+        int waitTime = (int)(nextSwitch - now);
+        DWORD res = WaitForSingleObject(g_wakeEvent, waitTime * 1000);
+        if (res == WAIT_OBJECT_0) {
+            if (g_exitFlag) break;
+            continue;
+        }
+
+        PCWSTR themePath = nextLight ? g_lightThemePath.c_str() : g_darkThemePath.c_str();
+        
+        if (*themePath) {
+            if (IsThemeApplied(themePath)) {
+                Wh_Log(L"[Theme] Theme already applied.");
+                continue;
+            }
+            ApplyTheme(themePath);
+        } else {
+            if (IsAppearanceApplied(nextLight ? light : dark)) {
+                Wh_Log(L"[Theme] Appearance already applied.");
+                continue;
+            }
+            ApplyAppearance(nextLight ? light : dark);
+        }
+    }
+    return 0;
+}
+
+void StartScheduler() {
+    if (g_timerThread) {
+        SetEvent(g_wakeEvent);
+        return;
+    }
+    g_wakeEvent = CreateEventW(nullptr, FALSE, FALSE, nullptr);
+    g_timerThread = CreateThread(nullptr, 0, ThemeScheduler, nullptr, 0, nullptr);
+}
+
+std::wstring TrimQuotes(const std::wstring& str) {
+    size_t start = 0;
+    size_t end = str.length();
+    if (!str.empty() && str.front() == L'"') start = 1;
+    if (end > start && str[end - 1] == L'"') end--;
+    return str.substr(start, end - start);
+}
+
+void LoadSettings() {
+    g_lightTime = ParseScheduleTime(Wh_GetStringSetting(L"Light"));
+    g_darkTime = ParseScheduleTime(Wh_GetStringSetting(L"Dark"));
+
+    auto rawLightPath = Wh_GetStringSetting(L"LightThemePath");
+    g_lightThemePath = rawLightPath ? TrimQuotes(rawLightPath) : L"";
+
+    auto rawDarkPath = Wh_GetStringSetting(L"DarkThemePath");

```

Comment:
`Wh_FreeStringSetting` needs to be called to free the strings that are returned by `Wh_GetStringSetting`.